open-metadata-implementation/adapters/open-connectors/event-bus-connectors/open-metadata-topic-connectors/kafka-open-metadata-topic-connector/src/main/java/org/odpi/openmetadata/adapters/eventbus/topic/kafka/KafkaOpenMetadataTopicConnector.java

```diff
@@ -363,6 +362,7 @@ public void disconnect() throws ConnectorCheckedException
         }
 
         try {
+            producer.safeCloseProducer();
```

Comment:
It would also be good to add a null check for `producer` here.

Reply:
The `producer` is also in the try-catch block

Reply:
I added the null check on producer.